policy/simulated/options.go

```diff
@@ -51,30 +51,83 @@ func NewOptionsFromRequest(r *http.Request) (Options, error) {
 // setDefaults sets any values for the options that were not intentionally set in the request body but which should have
 // consistent values for the length of the simulation, such as the created time for a comment or review.
 func (o *Options) setDefaults() {
-	now := time.Now()
 	for i, review := range o.AddReviews {
-		if review.CreatedAt.IsZero() {
-			review.CreatedAt = now
-		}
-
-		if review.LastEditedAt.IsZero() {
-			review.LastEditedAt = now
-		}
+		id := fmt.Sprintf("simulated-reviewID-%d", i)
+		sha := fmt.Sprintf("simulated-reviewSHA-%d", i)
 
-		review.ID = fmt.Sprintf("simulated-reviewID-%d", i)
-		review.SHA = fmt.Sprintf("simulated-reviewSHA-%d", i)
+		review.setDefaults(id, sha)
 		o.AddReviews[i] = review
 	}
 
 	for i, comment := range o.AddComments {
-		if comment.CreatedAt.IsZero() {
-			comment.CreatedAt = now
-		}
+		comment.setDefaults()
+		o.AddComments[i] = comment
+	}
+}
 
-		if comment.LastEditedAt.IsZero() {
-			comment.LastEditedAt = now
-		}
+type Comment struct {
+	CreatedAt    *time.Time `json:"created_at"`
+	LastEditedAt *time.Time `json:"last_edited_at"`
+	Author       string     `json:"author"`
+	Body         string     `json:"body"`
+}
 
-		o.AddComments[i] = comment
+// setDefaults sets the createdAt and lastEdtedAt values to time.Now() if they are otherwise unset
+func (c *Comment) setDefaults() {
+	now := time.Now()
+	if c.CreatedAt == nil {
+		c.CreatedAt = &now
+	}
+
+	if c.LastEditedAt == nil {
+		c.LastEditedAt = &now
+	}
+}
+
+func (c *Comment) toPullComment() *pull.Comment {
+	return &pull.Comment{
+		CreatedAt:    *c.CreatedAt,
+		LastEditedAt: *c.LastEditedAt,
+		Author:       c.Author,
+		Body:         c.Body,
+	}
+}
+
+type Review struct {
+	ID           string
+	SHA          string
+	CreatedAt    *time.Time `json:"created_at"`
+	LastEditedAt *time.Time `json:"last_edited_at"`
+	Author       string     `json:"author"`
+	Body         string     `json:"body"`
+	State        string     `json:"state"`
+	Teams        []string   `json:"teams"`
```

Comment:
Let's minimize the set of review fields we accept for now - we can always add more later, but taking away fields is harder. I think that means we should accept `CreatedAt,` `LastEditedAt`, `Author`, `Body`, and `State`. For everything else, let's either remove the field or set `json:"-"` so it is ignored when unmarshalling.